uber/config.py

```diff
@@ -73,9 +73,19 @@ def make_enum(self, enum_name, section, prices=False):
                 lookup[val] = desc
 
         enum_name = enum_name.upper()
-        setattr(self, enum_name + '_OPTS', opts)
-        setattr(self, enum_name + '_VARS', varnames)
-        setattr(self, enum_name + ('' if enum_name.endswith('S') else 'S'), lookup)
+        # When loading plugin's configs, we want to make sure we don't override existing enums
+        if hasattr(self, enum_name + '_OPTS'):
```

Comment:
So, I'm working on configdb where the configuration parsing code needs to
be completely separate from runtime - when the application starts the
config object will already be fully populated. Could we combine that here,
and just factor out ALL config population to a separate utility script?

On Wed, Oct 7, 2015 at 8:30 AM, Victoria Earl <notifications@github.com>
wrote:

> In uber/config.py
> <https://github.com/magfest/ubersystem/pull/1540#discussion_r41383622>:
>
> > @@ -73,9 +73,19 @@ def make_enum(self, enum_name, section, prices=False):
> >                  lookup[val] = desc
> >
> >          enum_name = enum_name.upper()
> > -        setattr(self, enum_name + '_OPTS', opts)
> > -        setattr(self, enum_name + '_VARS', varnames)
> > -        setattr(self, enum_name + ('' if enum_name.endswith('S') else 'S'), lookup)
> > +        # When loading plugin's configs, we want to make sure we don't override existing enums
> > +        if hasattr(self, enum_name + '_OPTS'):
>
> Yup. This is the relevant line in sideboard\config.py:
>
> for config_path in get_config_files(requesting_file_path, plugin):
>         # this gracefully handles nonexistent files
>         temp_config.merge(configobj.ConfigObj(config_path, encoding='utf-8', interpolation=False))
>
> I initially didn't want to mess with it, but it looks like we have enough
> complexity in what we're doing to config that it's probably better to take
> that approach.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/magfest/ubersystem/pull/1540/files#r41383622>.
>